File: boj/15954.cpp
#include <iostream>
#include <cmath>

using namespace std;

typedef double ld;

ld get_m(int K, int idx, int array[]){
    
    ld m = 0;
    for(int i =idx; i<idx+K; i++){
        m+=array[i];
    }

    return ld(m/K);
}

ld get_b(int K, int idx, int arr[], ld m){
    
    ld b=0;

    for(int i = idx; i < idx+K; i++){
        b += ((arr[i]-m)*(arr[i]-m));
    }

    b = b/K;

    return sqrt(b);
}

int main(){

    int N,K;
    ld minimum = 9999999;
    ld M, B;

    cin >> N >> K;

    int* dolls_like = new int[N];

    for(int i = 0; i<N; i++){
        cin >> dolls_like[i];
    }

    for(int _k = K; _k <= N; _k++){
        for(int idx = 0; idx<=N-_k; idx++){

            M = get_m(_k, idx, dolls_like);            
            B = get_b(_k, idx, dolls_like, M);
                        
            if(B<minimum){
                minimum=B;
            }
        }
    }

    printf("%.15f", minimum);
}

File: boj/15954.py
N,K = list(map(lambda x:int(x), input().split()))
dolls_like = list(map(lambda x : int(x), input().split()))

minimum = 99999999999999

for kk in range(K, N+1):
    for idx in range(kk, N+1):

        m = sum(dolls_like[idx-kk:idx])/kk
        v = 0
        for i in dolls_like[idx-kk:idx]:
            v += (i-m)**2
        b = (v / kk)**0.5

        if minimum > b:
            minimum = b


print(minimum)